api-server/src/models/categories.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const uuid = require('uuid/v4');
+
+const schema = {
+  id,
+  name: {type: String, uppercase: true, required: true},
+  display_name: {type: String, require: true, uppercase: true},
+  description: {type: String, required: false, enum:[no_description, description]},
+  category: {type: String, required: true}
+
+};
+
+class Categories {
+
+  constructor(name, display_name, description) {
+    this.database = []
+  }
+  
+  
+
+  get(_id) {
+    let  response = _id;
+    if (this.validate(entry)){
+      entry.id = uuid();
+      this.database.push(entry);
+      return entry;
+    }
+    else {return null}
+  }
+  
+  post(record) {
+    if (this.validate(record)){
+      record.id = uuid();
+      this.database.push(record);
+      return record;
+    }
+  }
+
+  put(_id, record) {
```

Comment:
You are really close to this, put has the same logic as post and git but put together. Reference the code you've already written!